lib/galaxy/tours/__init__.py

```diff
@@ -23,20 +26,24 @@ def tour_loader(contents_dict):
             step['orphan'] = True
         if title_default and 'title' not in step:
             step['title'] = title_default
-    return contents_dict
 
 
 class ToursRegistry:
     def __init__(self, tour_directories):
         self.tour_directories = util.config_directories_from_setting(tour_directories)
         self.load_tours()
 
-    def tours_by_id_with_description(self):
-        return [{'id': k,
-                 'description': self.tours[k].get('description', None),
-                 'name': self.tours[k].get('name', None),
-                 'tags': self.tours[k].get('tags', None)}
-                for k in self.tours.keys()]
+    def tours_by_id_with_description(self) -> TourList:
+        tours = []
+        for k in self.tours.keys():
+            tourdata = {
+                'id': k,
+                'description': self.tours[k].get('description'),
+                'name': self.tours[k].get('name'),
+                'tags': self.tours[k].get('tags')
+            }
+            tours.append(tourdata)
+        return parse_obj_as(TourList, tours)
```

Comment:
I think you can use here:
```suggestion
        return parse_obj_as(List[Tour], tours)
```
In case you want to use the `List` typing instead of a model for the list. But not 100% sure :)

Reply:
Unless I'm mistaken, we can't. That was my first approach, but I ran into the issue I mention in the PR description: this has to be serialized to json, and we cannot use pydantic's `model.json()` if it's a list of pydantic models: we can only serialize each model. To serialize the list you need to use `json.dumps()` (or our `safe_dumps` - same thing) - but `json.dumps()` doesn't know how to handle a model. Thus, using a pydantic model which is a list of models seems to be the most straightforward way to implement this. (It's different for the licenses endpoint where the response model is a list because we don't serialize that list of models, at least for now)